src/test/ui/hygiene/macro-metavars.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(rustc_attrs)]
+
+// run-pass
+
+#[rustc_macro_transparency = "transparent"]
+macro_rules! k {
+    ($($s:tt)*) => {
+        macro_rules! m {
+            ($y:tt) => {
+                $($s)*
+            }
+        }
+    }
+}
+
+k!(1 + $y);
+
+fn main() {
+    let x = 2;
+    assert_eq!(3, m!(x));
+}
```

Comment:
Could you add a test case making sure that macro variables are indeed hygienic?
I.e. a case that would be broken by https://github.com/rust-lang/rust/pull/69410.